apm/src/app/user/user.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { IUser } from './user';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserService {
+
+  currentUser: IUser | undefined;
```

Comment:
I would use `IUser | null` but that is my preference. Undefined in javascript is something that is not even defined, but we have defined the variable and so it should have some value. Again this is just a suggestion and something that I advocate.